rust/agama-lib/src/manager.rs

```diff
@@ -85,6 +88,72 @@ impl TryFrom<u32> for InstallationPhase {
     }
 }
 
+/// Finish method
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone, Copy, utoipa::ToSchema)]
+pub enum FinishMethod {
+    #[serde(rename = "halt")]
+    // Halt the system
+    Halt,
+    #[serde(rename = "reboot")]
+    // Reboots the system
+    Reboot,
+    #[serde(rename = "stop")]
+    Stop,
+    #[serde(rename = "poweroff")]
+    // Poweroff the system
+    Poweroff,
+}
+
+impl Default for FinishMethod {
+    fn default() -> Self {
+        Self::Reboot
+    }
+}
+
+impl fmt::Display for FinishMethod {
```

Comment:
We use the [strum](https://crates.io/crates/strum) crate for these conversions. You can check [the](https://github.com/agama-project/agama/blob/ca747cf59f5447fbc0cea0650400c3c626f28a41/rust/agama-lib/src/software/model/packages.rs#L35) [examples](https://github.com/agama-project/agama/blob/ca747cf59f5447fbc0cea0650400c3c626f28a41/rust/agama-lib/src/scripts/model.rs#L38).